packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1631,6 +1638,7 @@ export class ContainerRuntime
 			(path: string) => this.garbageCollector.isNodeDeleted(path),
 			new Map<string, string>(dataStoreAliasMap),
 			async (runtime: ChannelCollection) => provideEntryPoint,
+			this.blobContents,
```

Comment:
Passing blobs as separate thing is Ok, but I think it would be better if we could simple merge them into snapshot that we pass (first argument to ChannelCollection constructor). I.e. instead of passing ISnapshotTree, pass ISnapshot by creating it on the fly from base ISnapshot and all the properties it has (like sequenceNumber, blobs), but setting ISnapshot.snapshotTree to a sub-tree that the caller needs).
I.e., can we switch all the layers to work with ISnapshot? Where ISnapshot they receive is for the appropriate node in the tree?
The less concepts we have in the code - the better.

There will be back-compat considerations (as it might not be present at this layer). But I think we have to handle them (in different form) anyway. Where needed, we can use same patter as you use in other layers - i.e. accept ISnapshotTree | ISnapshot, and thus differentiate between legacy flow and more modern flow (with the goal of eventually switching all the code to modern flow).

What do you think?

Reply:
Added this support till datastorecontext level. We can extend this for dds later on as that is not required for now but at least we sow the seed for the same.